xapian-applications/omega/clickmodel/simplifieddbn.cc

```diff
@@ -0,0 +1,207 @@
+/** @file simplifieddbn.cc
+ * @brief SimplifiedDBN class - the Simplified DBN click model.
+ */
+/* Copyright (C) 2017 Vivek Pal
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <config.h>
+
+#include "simplifieddbn.h"
+
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+#define QID 0
+#define DOCIDS 1
+#define CLICKS 2
+
+string
+SimplifiedDBN::name()
+{
+    return "SimplifiedDBN";
+}
+
+static vector<string>
+get_docid_list(const vector<vector<string>> &sessions)
+{
+    vector<string> docids;
+    string docid;
+    for (size_t i = 0; i < sessions.size(); ++i) {
+	for (size_t j = 0; j <= sessions[i][DOCIDS].length(); ++j) {
+	    if (sessions[i][DOCIDS][j] != ',' && sessions[i][1][j] != '\0') {
```

Comment:
You're hard-coding `1` instead of `DOCIDS` in the second use here.

I'd factor out the character lookup, so it's clearer to the reader that it's the same character being tested each time and then appended, i.e.:

```
char ch = sessions[i][DOCIDS][j];
if (ch != ',' && ch != '\0') {
   docid += ch;
```